src/Microsoft.Health.Dicom.Metadata/Registration/Experimental/IBlobServiceBuilder.cs

```diff
@@ -0,0 +1,30 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using EnsureThat;
+using Microsoft.Health.Blob.Configs;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public interface IBlobServiceBuilder
+    {
+        IServiceCollection Services { get; }
+
+        BlobDataStoreConfiguration Configuration { get; }
+    }
+
+    internal sealed class BlobServiceBuilder : IBlobServiceBuilder
+    {
+        public IServiceCollection Services { get; }
+
+        public BlobDataStoreConfiguration Configuration { get; }
+
+        public BlobServiceBuilder(IServiceCollection serviceCollection, BlobDataStoreConfiguration config)
```

Comment:
is it a c# thing to have a constructor for a normal class inside of an Interface file?

Reply:
may be we should put this builder into a separated file.